pkg/kbld/cmd/package.go

```diff
@@ -48,6 +53,12 @@ func NewPackageCmd(o *PackageOptions) *cobra.Command {
 }
 
 func (o *PackageOptions) Run() error {
+	_, err := fmt.Fprintln(os.Stderr, `Command "package" is deprecated, please use 'imgpkg push', learn more in https://carvel.dev/imgpkg/docs/latest/commands/#push`)
```

Comment:
we got `logger := ctllog.NewLogger(os.Stderr)` a few lines below. let's use that to print this message. let's also prefix it with `Warning:`